apps/testingaccessibility/src/utils/handle-self-redeem.ts

```diff
@@ -2,22 +2,28 @@
  * handle self redeeming of a seat in bulk purchase
  * @param {string} email current user's email from session
  * @param {string} bulkCouponId current user's bulk coupon id from purchase
- * @param {(redeemPurchase) => void} onSuccess callback to be called when purchase is redeemed
+ * @param {(params) => void} callback function to be called after attempting coupon redemption
```

Comment:
Replaces `onSuccess` with generic `callback` handler which responds with either `status: 'success'` or `status: 'failed'`. The caller than either has the purchase object on success or an error message on failure and can handle those scenarios appropriately.